### PR TITLE
[Term Entry] D3 Interactivity .on()

### PR DESCRIPTION
**Pull request by kayomi-codes**
### Description

- Added the .on() term for D3
- Closes issue #3882

This is my first time contributing to Codecademy Docs and I'm also new to programming, so I will genuinely appreciate all constructive feedback on my PR. Thank you very much in advance.

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

**Review by dakshdeepHERE — CHANGES_REQUESTED** [review]
I have requested few changes from my side for you @kayomi-codes

File: content/d3/concepts/interactivity/terms/on/on.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+```pseudo
+selection.on(event type, callback function)
+```
+
+The **event type** is a *string* that specifies the DOM event type used, and the [callback function](https://www.codecademy.com/resources/docs/javascript/callbacks) is the *function* that is called when the event is triggered.
+
+## Example
+
+In the following code snippet, hovering over a square makes it turn yellow:
```
[the remark this answers is not in the document]
Comment (kayomi-codes) [reply content/d3/concepts/interactivity/terms/on/on.md:31]: Sure! That sounds perfect to represent what the code snippet does more clearly. I will add an image and modify the rest of the suggested changes you sent me as soon as I can.

**Comment by kayomi-codes** [comment]
I have added your suggested changes and attached an image as output for the code snippet @dakshdeepHERE

**Comment by dakshdeepHERE** [comment]
Hey @kayomi-codes, the output image is looking perfect. I will go through the entry again now and we will move to further reviews. 🚀

**Comment by kayomi-codes** [comment]
Great! Thank you for your time.

**Review by dakshdeepHERE — APPROVED** [review]
Looks good for second review.

File: content/d3/concepts/interactivity/terms/on/on.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+In **D3.js**, the **d3.selection.on()** function is used to add a specific event handler to an element. An event can be any Document Object Model (DOM) event type such as `click`, `zoom`, `mouseenter`, `mouseleave`, `mouseover`, `mouseout`, and `mousemove`.
+
+## Syntax
+
+```pseudo
+selection.on(event type, callback function)
+```
+
+- `event type`: The **event type** is a *string* that specifies the DOM event type used.
```
Comment (SSwiniarski) [inline content/d3/concepts/interactivity/terms/on/on.md:25]:
```suggestion
- `event type`: The event type is a string that specifies the DOM event type used.
```

File: content/d3/concepts/interactivity/terms/on/on.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+- `callback function`: The [callback function](https://www.codecademy.com/resources/docs/javascript/callbacks) is the *function* that is called when the event is triggered.
```
Comment (SSwiniarski) [inline content/d3/concepts/interactivity/terms/on/on.md:26]:
```suggestion
- `callback function`: The [callback function](https://www.codecademy.com/resources/docs/javascript/callbacks) is the function that is called when the event is triggered.
```

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@kayomi-codes I did a second review and had a few comments.

File: content/d3/concepts/interactivity/terms/on/on.md (hunk 3 above)
Comment (kayomi-codes) [reply content/d3/concepts/interactivity/terms/on/on.md:26]: Got it. I will add all the suggested changes as soon as possible.

**Comment by kayomi-codes** [comment]
I have added the suggested changes @SSwiniarski

**Review by SSwiniarski — APPROVED** [review]
@kayomi-codes Looks good.

**Pull request merged**